draft-gpew-priv-ppm.md

```diff
@@ -900,37 +900,70 @@ The named parameters are:
+until the collector provides the aggregation parameter in the `CollectReq`. For
+those reasons, collect requests are handled asynchronously.
+
+Upon receipt of a `CollectReq`, the leader begins by checking that the request
+meets the requirements of the batch parameters using the procedure in
+{{batch-parameter-validation}}. If so, it immediately sends the collector a
+response with HTTP status 202 Accepted. The body consists of a `CollectResp`:
 
-It begins by checking that the request meets the requirements of the batch
-parameters using the procedure in {{batch-parameter-validation}}. If so, it
-obtains the helper's encrypted aggregate share for the batch interval by sending
-an AggregateShareReq to the helper as described in {{aggregate-share-request}}.
-(This request may too have been made in advance.)
-
-Next, the leader computes its own aggregate share by aggregating all of the
-verified output shares that fall within the batch interval. Finally, it responds
-with HTTP status 200 and a body consisting of a CollectResp message:
-
-[OPEN ISSUE: What happens if this all takes a really long time.]
+~~~
+struct {
+  Url collect_job;
+} CollectResp;
+~~~
 
-[TODO: Decide if and how the collector's request is authenticated.]
+`CollectResp.collect_job` is a URL identifying the collect job that can be
+polled by the collector. It SHOULD contain an unpredictable component to
+prevent enumeration attacks against a leader's collect results. The HTTP
+response MAY include a Retry-After header field to suggest a polling interval to
+the client.
+
+The leader then begins working with the helper to prepare the shares falling
+into `CollectReq.batch_interval` (or continues this process, depending on the
+VDAF) as described in {{pa-aggregate}}. After receiving a `CollectResp`, the
+collector periodically makes HTTP GET requests to `CollectResp.collect_job` to
+check on the status of the collect job and eventually obtain the result. If the
+collect job is not finished yet, the leader responds with HTTP status 202
+Accepted and a `CollectResp`. The `CollectResp` MAY contain a different
+`collect_job` than previous `CollectResp`s and the HTTP response MAY include a
+`Retry-After` header field to suggest a polling interval to the collector.
+
+Once all the necessary reports have been prepared, the leader obtains the
+helper's encrypted aggregate share for the batch interval by sending an
+`AggregateShareReq` to the helper as described in {{aggregate-share-request}}.
+The leader then computes its own aggregate share by aggregating all of the
+prepared output shares that fall within the batch interval.
+
+When both aggregator's shares are successfully obtained, then the leader
+responds to subsequent HTTP GET requests to `CollectResp.collect_job` with HTTP
+status 200 OK and a body consisting of a `CollectResult`:
 
 ~~~
 struct {
-  EncryptedAggregateShare shares<1..2^16-1>;
-} CollectResp;
+  EncryptedAggregateShare shares<1..2^24-1>;
```

Comment:
Why the size increase here? It seems unlikely to me that we'll ever have aggregate shares larger than 2^16 - 1 bytes.